### PR TITLE
Make FindZLIB consistent with official definitions

### PR DESCRIPTION
**Pull request by xkszltl**
CMake 3 already has FindZLIB.
[https://cmake.org/cmake/help/v3.13/module/FindZLIB.html](https://cmake.org/cmake/help/v3.13/module/FindZLIB.html)

### CODE REVIEW

**Comment by tchaikov** [comment]
rocksdb claims to support cmake 2.6. and yeah, FindZLIB is also available in 2.6. see https://cmake.org/cmake/help/cmake2.6docs.html#module:FindZLIB . so i'd suggest remove `Findzlib.cmake` instead.

**Comment by xkszltl** [comment]
@tchaikov 
Aha! I see what's going on.
2.6's FindZlib uses `ZLIB_INCLUDE_DIR`, which is the same as what rocksdb uses currently, while 3.x uses `ZLIB_INCLUDE_DIRS`.
Will add an alias or version-based condition to address it.

Thanks for the link

**Comment by xkszltl** [comment]
Updated base on @tchaikov suggestion.

File: CMakeLists.txt [hunk 1]
```diff
@@ -94,7 +94,11 @@ else()
   if(WITH_ZLIB)
     find_package(zlib REQUIRED)
     add_definitions(-DZLIB)
-    include_directories(${ZLIB_INCLUDE_DIR})
+    if(CMAKE_VERSION VERSION_LESS "3.0")
```
Comment (tchaikov) [inline CMakeLists.txt]: @xkszltl i'd suggest avoid checking the `CMAKE_VERSION` unless we have no other options. as it's a magic number in this context. how about just checking if `ZLIB_INCLUDE_DIRS` is available, and use `ZLIB_INCLUDE_DIR` as a fallback? like
```cmake
if(ZLIB_INCLUDE_DIRS)
  include_directories(${ZLIB_INCLUDE_DIRS})
else()
  include_directories(${ZLIB_INCLUDE_DIR})
endif()
```

and Happy New Year!

File: CMakeLists.txt [hunk 2]
```diff
@@ -94,7 +94,13 @@ else()
   if(WITH_ZLIB)
     find_package(zlib REQUIRED)
     add_definitions(-DZLIB)
-    include_directories(${ZLIB_INCLUDE_DIR})
+    if (DEFINED ZLIB_INCLUDE_DIRS)
```
Comment (tchaikov) [inline CMakeLists.txt]: please note cmake also consider a condition also false if `ZLIB_INCLUDE_DIRS` is not defined. so for simplicity and consistency, i'm inclined to just put  `if(ZLIB_INCLUDE_DIRS)`. and to be more consistent with other part of the cmake scripts in this project, could you remove the space after `if`?

Comment (xkszltl) [reply CMakeLists.txt]: Good to know that!

**Pull request closed without merging**